libraries/rodeos/wasm_ql.cpp

```diff
@@ -0,0 +1,842 @@
+      db_view_state db_view_state{state_account, *thread_state.shared->db, write_session,
+                                  contract_kv_prefix};
+
+      std::string result = "{\"server_type\":\"wasm-ql\"";
+
+      {
+         global_property_kv table{{db_view_state}};
+         bool found = false;
+         if (table.primary_index.begin() != table.primary_index.end())
+         {
+            auto record = table.primary_index.begin().value();
+            if (auto* obj = std::get_if<ship_protocol::global_property_v1>(&record))
+            {
+               found = true;
+               result += ",\"chain_id\":" + eosio::convert_to_json(obj->chain_id);
+            }
+         }
+         if (!found)
+            throw std::runtime_error("No global_property_v1 records found; is filler running?");
+      }
+
+      {
+         fill_status_sing sing{state_account, db_view_state, false};
+         if (sing.exists())
+         {
+            std::visit(
+                [&](auto& obj) {
+                   result += ",\"head_block_num\":\"" + std::to_string(obj.head) + "\"";
+                   result += ",\"head_block_id\":" + eosio::convert_to_json(obj.head_id);
+                   result += ",\"last_irreversible_block_num\":\"" +
+                             std::to_string(obj.irreversible) + "\"";
+                   result += ",\"last_irreversible_block_id\":" +
+                             eosio::convert_to_json(obj.irreversible_id);
+                },
+                sing.get());
+         }
+         else
+            throw std::runtime_error("No fill_status records found; is filler running?");
+      }
+
+      result += "}";
+
+      thread_state.action_return_value.assign(result.data(), result.data() + result.size());
+      return thread_state.action_return_value;
+   }
+
+   struct get_block_params
+   {
+      std::string block_num_or_id = {};
+   };
+
+   EOSIO_REFLECT(get_block_params, block_num_or_id)
+
+   const std::vector<char>& query_get_block(wasm_ql::thread_state& thread_state,
+                                            const std::vector<char>& contract_kv_prefix,
+                                            std::string_view body)
+   {
+      get_block_params params;
+      std::string s{body.begin(), body.end()};
+      eosio::json_token_stream stream{s.data()};
+      try
+      {
+         from_json(params, stream);
+      }
+      catch (std::exception& e)
+      {
+         throw std::runtime_error("An error occurred deserializing get_block_params: "s + e.what());
+      }
+
+      rocksdb::ManagedSnapshot snapshot{thread_state.shared->db->rdb.get()};
+      chain_kv::write_session write_session{*thread_state.shared->db, snapshot.snapshot()};
+      db_view_state db_view_state{state_account, *thread_state.shared->db, write_session,
+                                  contract_kv_prefix};
+
+      std::string bn_json = "\"" + params.block_num_or_id + "\"";
```

Comment:
`from_json` will see it as a shortened string, followed by extra stuff at the end that it won't attempt to parse. Adding a todo to extend abieos's conversions and use those instead of json parsing.